### PR TITLE
Reviewer Rob - Daemonize ourselves when started by start-stop-daemon

### PR DESCRIPTION
**Pull request by bossmc**
This is the sprout version of the fix for https://github.com/Metaswitch/cpp-common/issues/473 (cpp-common PR coming shortly) if you're happy I'll port to other C++ projects.

I'm testing live in parallel with your code review (as the changes are small but high risk :grinning:)

### CODE REVIEW

File: debian/bono.init.d [hunk 1]
```diff
@@ -202,7 +200,7 @@ do_start()
         # daemon is not running, so attempt to start it.
         setup_environment
         get_daemon_args
-        $namespace_prefix start-stop-daemon --start --quiet --background --pidfile $PIDFILE --exec $DAEMON --chuid $NAME --chdir $HOME -- $DAEMON_ARGS \
+        /usr/share/clearwater/bin/run-in-signaling-namespace start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON --chuid $NAME --chdir $HOME -- $DAEMON_ARGS --daemon --pidfile=$PIDFILE \
```
Comment (rkd-msw) [inline debian/bono.init.d]: If `--daemon --pidfile=$PIDFILE` are daemon args, put them in $DAEMON_ARGS!

Comment (rkd-msw) [inline debian/bono.init.d]: Or at least do something like $RUN_ARGS.

File: debian/bono.init.d [hunk 2]
```diff
@@ -234,7 +232,7 @@ do_run()
 
         setup_environment
         get_daemon_args
-        $namespace_prefix start-stop-daemon --start --quiet --exec $DAEMON --chuid $NAME --chdir $HOME -- $DAEMON_ARGS \
+        /usr/share/clearwater/bin/run-in-signaling-namespace start-stop-daemon --start --quiet --exec $DAEMON --chuid $NAME --chdir $HOME -- $DAEMON_ARGS \
```
Comment (rkd-msw) [inline debian/bono.init.d]: Should this stll have `--pidfile $PIDFILE`? Maybe for Docker?

**Comment by rkd-msw** [comment]
Cool.

File: debian/bono.init.d (hunk 2 above)
Comment (bossmc) [inline debian/bono.init.d]: `supervisord` doesn't look at the pidfile.

File: debian/bono.init.d (hunk 1 above)
Comment (bossmc) [inline debian/bono.init.d]: These args control how the process integrates with it's launcher, not how it should function when it's running.  I don't think they therefore belong in `$DAEMON_ARGS`.

**Pull request merged**